packages/backend/src/helpers/global-variable.ts

```diff
@@ -74,6 +73,15 @@ const globalVariable = async (
       },
     },
     pushTriggerItem: (triggerItem: ITriggerItem) => {
+      if ($.execution.testRun) {
+        $.triggerOutput.data.push(triggerItem);
+        throw new EarlyExitError();
+      }
+
+      if (isAlreadyProcessed(triggerItem.meta.internalId)) {
+        throw new EarlyExitError();
+      }
+
       $.triggerOutput.data.push(triggerItem);
```

Comment:
```suggestion
      if (isAlreadyProcessed(triggerItem.meta.internalId) && !$.execution.testRun) {
        // early exit as we do not want to process duplicate items in actual executions
        throw new EarlyExitError();
      }

      $.triggerOutput.data.push(triggerItem);

      if ($.execution.testRun) {
        // early exit after receiving one item as it is enough for test execution
        throw new EarlyExitError();
      }
```

While we may keep the implementation as-is, it may be good to write some explanations here, like above.